backend/src/dao/controller/task.controller.ts

```diff
@@ -0,0 +1,52 @@
+import { Body, Controller, Delete, Get, HttpCode, Param, Post, Put } from '@nestjs/common'
+import { ApiBearerAuth, ApiTags } from '@nestjs/swagger'
+import { Role } from 'src/user/user.dto'
+import { CreateTaskDto, TaskDto, TaskQueryDto, UpdateTaskDto } from '../../task/task.dto'
+import { TaskService } from '../../task/task.service'
+import { DaoAuth } from '../dao.auth.decorator'
+
+@ApiBearerAuth()
+@ApiTags('task')
+@Controller('/dao/:daoId/task')
+export class TaskController {
+  constructor(private taskService: TaskService) {}
+
+  @Post('find')
+  @HttpCode(200)
+  @DaoAuth()
+  find(@Param('daoId') daoId: string, @Body() query: TaskQueryDto): Promise<TaskDto[]> {
```

Comment:
If I understand this code correctly, tasks are associated with DAOs. Shouldn't tasks be associated with projects instead?

DAO -> Project -> Task

Reply:
Yes the structure is like the above DAO > Project > Task
To simplify a bit the authorization I opted to require explicity the daoId from controller (used to load the members and roles see [@DaoAuth](https://github.com/opny721/sweat-token/blob/feature/39-people-section/backend/src/dao/dao.auth.decorator.ts#L8)). When we have a stable flow, we can improve the API interfaces